api/types.proto

```diff
@@ -277,6 +277,50 @@ message UpdateConfig {
 
 	// Amount of time between updates.
 	Duration delay = 2 [(gogoproto.nullable) = false];
+
+	enum FailureAction {
+		PAUSE = 0;
+		CONTINUE = 1;
+		// TODO(aaronl): Add ROLLBACK as a supported failure mode.
+		// (#486)
+	}
+
+	// FailureAction is the action to take when an update failures.
+	// Currently, a failure is defined as a single updated task failing to
+	// reach the RUNNING state. In the future, there will be configuration
+	// to define what is treated as a failure (see #486 for a proposal).
+	FailureAction failure_action = 3;
+}
+
+// UpdateStatus is the status of an update in progress.
+message UpdateStatus {
+	enum UpdateState {
+		NEW = 0;
+		UPDATING = 1;
+		PAUSED = 2;
+		COMPLETED = 3;
+		// TODO(aaronl): add ROLLING_BACK, ROLLED_BACK as part of
+		// rollback support.
+	}
+
+	// State is the state of this update. It indicates whether the
+	// update is in progress, completed, or is paused.
+	UpdateState state = 1;
+
+	// Started is the time at which the update was started.
+	// TODO(aaronl): Consider adding a timestamp showing when the most
+	// recent task update took place. Currently, this is nontrivial
+	// because each service update kicks off a replacement update, so
+	// updating the service object with a timestamp at every step along
+	// the rolling update would cause the rolling update to be constantly
+	// restarted.
+	Timestamp started = 2;
```

Comment:
Should we have a `completed` as well?

Just thinking about the general workflow:
- Create a service, inspect will show UpdateStatus as null
- Start an update: at first, UpdateStatus will be null but eventually it will start
- inspect will show `state=UPDATING` with a started timestamp
- after a while, `state=COMPLETED and again the the started timestamp
- This will remain forever on the service, until we perform another update

Since most services will always have a `UpdateStatus` field set to something, maybe we should make it more useful by having the time of completion.

Then again, we can always add it later so this shouldn't block this PR, just something to keep in mind